lib/statsd/instrument/client.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require 'statsd/instrument/datagram'
+require 'statsd/instrument/datagram_builder'
+require 'statsd/instrument/statsd_datagram_builder'
+require 'statsd/instrument/dogstatsd_datagram_builder'
+require 'statsd/instrument/null_sink'
+require 'statsd/instrument/udp_sink'
+require 'statsd/instrument/capture_sink'
+require 'statsd/instrument/log_sink'
+
+class StatsD::Instrument::Client
+  attr_reader :sink, :datagram_builder_class, :prefix, :default_tags, :sampling_disabled, :default_sample_rate
+
+  def initialize(
+    sink: StatsD::Instrument::NullSink.new,
+    prefix: nil,
+    sampling_disabled: false,
+    default_sample_rate: 1,
+    default_tags: nil,
+    datagram_builder_class: StatsD::Instrument::Environment.datagram_builder_class
+  )
+    @sink = sink
+    @datagram_builder_class = datagram_builder_class
+
+    @prefix = prefix
+    @default_tags = default_tags
+    @default_sample_rate = default_sample_rate
+    @sampling_disabled = sampling_disabled
+  end
+
+  # Emits a counter metric.
+  # @param name [String] The name of the metric.
+  # @param value [Integer] (default: 1) The value to increment the counter by.
+  #
+  #   You should not compensate for the sample rate using the counter increment. E.g., if
+  #   your sample rate is 0.01, you should <b>not</b> use 100 as increment to compensate for it.
+  #   The sample rate is part of the packet that is being sent to the server, and the server
+  #   should know how to handle it.
+  #
+  # @param sample_rate [Float] (default: nil)
+  # @param tags [Hash, Array] (default: nil)
+  # @return StatsD::Instrument::Datagram
```

Comment:
looks like you forgot to update the documentation to show that we are changing to fire and forget style methods

```suggestion
  # @return [void]
```